### DISCUSSION TITLE
Feat request: redeliver with suppression list ignore

### DISCUSSION THREAD

**Discussion by arthurzenika** [postalserver/postal#2809] [Feature suggestions]
We have something that happens quite often on our postal instances : something is wrong on the recipient side (over quota, smtp migration, ssl wrongly configured, etc) so postal rightfully goes into soft fail/hard fail and the email ends up on the suppression list. So far so good. That's a great feature. 

But... when the recipient side fixes the problem and we are still in the 30 days "pause" period, we want to redeliver a message or even a test message, but that will be intercepted by the suppression list. What we do is we go and do an SQL query `delete from  suppressions where address = 'user@example.org';` and *then* go and click "Redeliver message". That seems to fix the problem and we can then (if the older messages are still in the database) go back and redeliver a bunch of failed messages.  

The feature could look like : having an "advanced" button (hidden somewhere) that enables a "Redeliver message while ignoring suppression list".

**Comment by willpower232** [comment]
I think the way to remove someone from the suppression list is to release a held message to them. A message that was held because they were on the suppression list.

It may in fact be a bug if you cannot click on redeliver on an already failed message in order to take them off the list.

**Reply by arthurzenika** [reply]
I can click on Redeliver but it seems to go strait to Error claiming the user is in the suppression list. Indeed, that could be a bug.

**Reply by adamcooke** [reply]
As Will says, a user is removed from the suppression list when mail is successfully delivered to them. If a message, which is currently in a SoftFail state is retried manually through the UI, the recipient will be removed from the suppression list. 

I have quickly tried to reproduce this locally but it seems fine to me.

**Reply by arthurzenika** [reply]
I can confirm that when a user is "manually" removed (via SQL) from the suppression list and that some emails succeed in being sent out, after bit it is reintroduced into the suppression list. Is there a error count somewhere that should be reset too ?

**Reply by arthurzenika** [reply]
Maybe the previous emails can be re-sent, but new emails are blocked by counting the number of previous errors before attempting to send out the email and so the email address ends up in the suppression list again ?

**Reply by willpower232** [reply]
Its probably something like that, unfortunately if someone is having issues receiving emails then there isn't much you can do.

**Reply by arthurzenika** [reply]
In my use case the destination email is fixed, I have some retried emails that are in the "Sent" status and yet new email go direct to Error and the email is reintroduced in the Suppression list.

**Reply by willpower232** [reply]
what is the error message on the new email?

**Reply by arthurzenika** [reply]
So on this we can see the message at 16:54 is sent correctly (redelivered, so removed from the suppression list)

![image](https://github.com/postalserver/postal/assets/445200/696c7f75-55ad-4732-b04c-432d5a76e341)

But the new message at 18:07 goes straight to error : 

![image](https://github.com/postalserver/postal/assets/445200/43026879-1026-451b-a449-d5b34db72ee9)

The "Show technical details" is the : 

```
FrozenError: can't modify frozen String: ""

Support Ref: J-ZIMOUBRJES
```

**Reply by willpower232** [reply]
Do you need to update Postal to resolve the frozen string error? I can't remember where that got fixed off the top of my head.

**Reply by arthurzenika** [reply]
Unfortunately upgrading that environment is a bit costly (but soon the underlying infrastructure will be upgraded which will help). Can I pick the error message in the logs ? Do you think this behavior is specific to that version (2.1.4) ?

**Reply by adamcooke** [reply]
Yes. Upgrading should resolve the issue.

**Reply by arthurzenika** [reply]
Can you point me to some code so that I understand the issue and how it was fixed ? (thanks a lot for taking the time answer my questions)

**Reply by arthurzenika** [reply]
We're looking into the impacts of upgrading. Would upgrading to 2.3.2 be enough ? or do we have to upgrade 3.x ?

**Reply by sandrovenezuela** [reply]
Hi @arthurzenika

Did you manage to update Postal from version 2.1.4 to 2.3.2?

**Reply by arthurzenika** [reply]
hi @sandrovenezuela we haven't decided yet if we were to upgrade to 2.3.2 for now or if we have to upgrade to v3 to see a fix to the problem described here. Without further news or info, we will probably upgrade to 2.3.2 in the following days.

**Comment by sandrovenezuela** [comment]
Hi @arthurzenika,

Thank you for the return.

I tried to update my Postal server from version 2.1.4 to 2.3.2 and I couldn't. When executing the command "postal upgrade 2.3.2" the Docker images were not updated.

Postal continued to work, but in version 2.1.4.

If you could share how the update went, if everything went as expected, that would be great.

Thanks!

**Comment by abhilashtekiestech** [comment]
@arthurzenika  I had the same issue, I resolved it by updating the config parameter default_suppression_list_automatic_removal_days to 0, I don't know if this is present in older version but maybe this will help you
